// react-native-connect-tracker/ConnectTrackerEvent.js
import { Platform } from 'react-native'

class ConnectTrackerEvent {
    constructor(data = {}) {
        this.name = data.name;
        this.trackingId = data.trackingId;
        this.sdkVersion = data.sdkVersion;
        this.impressionId = data.impressionId;
        this.timestamp = data.timestamp;
        this.value = data.value ? data.value : null;
        this.advertisingId = data.advertisingId ? data.advertisingId : null;
        this.idfa = data.idfa ? data.idfa : null;
    
        console.log('ConnectTrackerEvent', JSON.stringify(data), JSON.stringify(this));
    }

    toParameterObject() {
        const parameters = {
            CI_TRACKING_ID: this.trackingId,
            CI_CONNECTSDK_VERSION: this.sdkVersion,
            CI_IMPRESSION_ID: this.impressionId,
            CI_TIMESTAMP: this.timestamp
        };

        if (this.value && this.value.length > 0) {
            parameters.CI_VALUE = this.value;
        }

        if (Platform.OS === 'android') {
            parameters.CI_ADVERTISING_ID = this.advertisingId;
        } else if (Platform.OS === 'ios') {
            parameters.CI_IDFA = this.idfa;
        }

        return parameters;
    }
}

export default ConnectTrackerEvent;
